lib/api/bucketDelete.js

```diff
@@ -31,17 +31,17 @@ export default function bucketDelete(authInfo, request, log, cb) {
         log,
     };
 
-    return services.metadataValidateAuthorization(metadataValParams,
+    return metadataValidateBucketAndObj(metadataValParams, log,
```

Comment:
for methods where we are only validating the bucket, I thought it would make more sense to use a function that only retrieves the bucket in MD and does bucket authorization. If you agree, you can add the following two functions in `metadataUtils` (I removed them in the versioning PR since we weren't using them yet):

```
/** metadataGetBucket - retrieves bucket from metadata, returning error if
 * bucket is shielded
 * @param {string} requestType - type of request
 * @param {string} bucketName - name of bucket
 * @param {RequestLogger} log - request logger
 * @param {function} cb - callback
 * @return {undefined} - and call callback with err, bucket md
 */
export function metadataGetBucket(requestType, bucketName, log, cb) {
    return metadata.getBucket(bucketName, log, (err, bucket) => {
        if (err) {
            log.debug('metadata getbucket failed', { error: err });
            return cb(err);
        }
        if (bucketShield(bucket, requestType)) {
            log.debug('bucket is shielded from request', {
                requestType,
                method: 'metadataGetBucketAndObject',
            });
            return cb(errors.NoSuchBucket);
        }
        log.trace('found bucket in metadata');
        return cb(null, bucket);
    });
}

/** metadataValidateBucket - retrieve bucket from metadata and check if user
 * is authorized to access it
 * @param {object} params - function parameters
 * @param {AuthInfo} params.authInfo - AuthInfo class instance, requester's info
 * @param {string} params.bucketName - name of bucket
 * @param {string} params.requestType - type of request
 * @param {RequestLogger} log - request logger
 * @param {function} callback - callback
 * @return {undefined} - and call callback with params err, bucket md
 */
export function metadataValidateBucket(params, log, callback) {
    const { authInfo, bucketName, requestType } = params;
    const canonicalID = authInfo.getCanonicalID();
    return metadataGetBucket(requestType, bucketName, log, (err, bucket) => {
        if (err) {
            return callback(err);
        }
        // still return bucket for cors headers
        if (!isBucketAuthorized(bucket, requestType, canonicalID)) {
            log.debug('access denied for user on bucket', { requestType });
            return callback(errors.AccessDenied, bucket);
        }
        return callback(null, bucket);
    });
}
```